File: Java/Array/MoveNegative/MoveNegative.java
public class MoveNegative {

    public static void shiftIntegers(int[] arr, int n) {
        int j = 0, temp;

        for (int i = 0; i < n; i++) {
            if (arr[i] < 0) {
                if (i != j) {
                    temp = arr[i];
                    arr[i] = arr[j];
                    arr[j] = temp;
                }
                j++;
            }
        }
    }

    public static void shiftAll(int[] arr, int left, int right) {
        // loop to iterate over the array from left to right
        while (left <= right) {
            if (arr[left] < 0 && arr[right] < 0) {
                left++;
            } else if (arr[left] > 0 && arr[right] < 0) {
                int temp = arr[left];
                arr[left] = arr[right];
                arr[right] = temp;
                right--;
            } else if (arr[left] > 0 && arr[right] > 0) {
                right--;
            } else {
                left++;
                right--;
            }

        }
    }

    public static void printArray(int arr[], int n) {
        for (int i = 0; i < n; i++)
            System.out.print(arr[i] + " ");
    }

    public static void main(String args[]) {
        int arr[] = { 2, 4, -10, 13, -7, -60, 52, 8, -19 };
        int n = arr.length;
        // shiftIntegers(arr, n);
        shiftAll(arr, 0, arr.length - 1);
        printArray(arr, n);
    }

}

File: Javascript/Array/MoveNegative/MoveNegative.js
const moveNegative = (arr) => {
  let negArr = [];
  let posArr = [];

  for (let i = 0; i < arr.length - 1; i++) {
    if (arr[i] < 0) {
      negArr.push(arr[i]);
    } else {
      posArr.push(arr[i]);
    }
  }
  let outArr = [...negArr, ...posArr];

  console.log("output array", outArr);
};

function rearrange(arr, n) {
  let j = 0;
  for (let i = 0; i < arr.length; i++) {
    if (arr[i] < 0) {
      if (i != j) {
        let temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
      }
      j++;
    }
  }
  console.log(arr);
}

let arr = [-2, 3, -4, 6, -1, 45, -56, 6];

rearrange(arr);
// moveNegative(arr);
